sw/device/tests/dif/dif_aon_timer_smoketest.c

```diff
@@ -0,0 +1,72 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+//#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_aon_timer.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+const test_config_t kTestConfig;
+
+bool test_main(void) {
+  dif_aon_timer_t aon;
+
+  LOG_INFO("Running AON timer test");
+
+  // Initialise AON Timer.
+  dif_aon_timer_params_t params = {
+      .base_addr = mmio_region_from_addr(TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR),
+  };
+  CHECK(dif_aon_timer_init(params, &aon) == kDifAonTimerOk);
+
+  // Make sure that wake-up timer is stopped and IRQs cleared.
+  CHECK(dif_aon_timer_wakeup_stop(&aon) == kDifAonTimerOk);
+
+  CHECK(dif_aon_timer_irq_acknowledge(&aon, kDifAonTimerIrqWakeupThreshold) ==
+        kDifAonTimerOk);
+
+  bool is_pending;
+  CHECK(dif_aon_timer_irq_is_pending(&aon, kDifAonTimerIrqWakeupThreshold,
+                                     &is_pending) == kDifAonTimerOk);
+  CHECK(is_pending == false);
+
+  // Test the wake-up timer functionality by setting a single cycle counter.
+  CHECK(dif_aon_timer_wakeup_start(&aon, 1, 0) == kDifAonTimerOk);
+
+  // Delay to compensate for AON Timer 200kHz clock (10uS should suffice).
+  usleep(100);
+  CHECK(dif_aon_timer_irq_is_pending(&aon, kDifAonTimerIrqWakeupThreshold,
+                                     &is_pending) == kDifAonTimerOk);
+  CHECK(is_pending == true);
+
+  CHECK(dif_aon_timer_wakeup_stop(&aon) == kDifAonTimerOk);
+
+  // Make sure that watchdog timer is stopped and IRQs cleared.
+  CHECK(dif_aon_timer_watchdog_stop(&aon) == kDifAonTimerWatchdogOk);
+
+  CHECK(dif_aon_timer_irq_acknowledge(
+            &aon, kDifAonTimerIrqWatchdogBarkThreshold) == kDifAonTimerOk);
+
+  CHECK(is_pending == false);
```

Comment:
I don't think this is right - is_pending is still set from above. Also the Wkup IRQ never gets acknowledged.

Reply:
Nice catch!

Done.